### DISCUSSION TITLE
How to fix error with fonts "magick: unable to read font"?

### DISCUSSION THREAD

**Discussion by gerkas** [ImageMagick/ImageMagick#6995] [Help]
### ImageMagick version

7.1.1-25

### Operating system, version and so on

WSL Debian

### Description

Hi, I'm trying to write some text but I getting this error:

```
$ magick -background lightblue -fill blue -font arial -pointsize 72 label:Anthony label.gif

magick: unable to read font arial @ warning/annotate.c/RenderType/1024.
magick: unable to read font arial @ error/annotate.c/RenderFreetype/1654.
```

The commands below doesn´t list anything.

```
$ convert -list font
$ identify -list font
$
```

And this others prints
```
$ ls /usr/local/share/fonts/
$ ls /etc/fonts
conf.avail  conf.d  fonts.conf
```

Current IM version. Thanks for any help.
```
$ magick -version
Version: ImageMagick 7.1.1-25 Q16-HDRI x86_64 21883 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5)
Delegates (built-in): fftw freetype jbig jng jpeg lcms lzma png tiff webp xml zlib zstd
Compiler: gcc (12.2)
```

**Comment by dlemstra** [comment]
It looks like your version of ImageMagick is build without fontconfig support?

**Reply by gerkas** [reply]
It could be. I installed latest version since it happens the same with 7.1.1-24. How can I fix this issue?

**Reply by anko20094** [reply]
The same issue. Did you fix it?

**Reply by GeeMack** [reply]
If ImageMagick is unable to find an installed font, you can still use it in a command by using its path and filename: `... -font "path\to\font\arial.tff" ...`

**Comment by fmw42** [comment]
What do you get from

magick -list font

If you want to include all your fonts on your system, see the script https://imagemagick.org/Usage/scripts/imagick_type_gen

**Comment by nodecentral** [comment]
I have a similar issue, what is the best way to handle this, 

````
magick: unable to read font `Roboto-Bold' @ warning/annotate.c/RenderType/1024.
magick: delegate library support not built-in '/usr/share/fonts/opentype/urw-base35/NimbusSans-Regular.otf' (Freetype) @ warning/annotate.c/RenderFreetype/2113.
magick: no decode delegate for this image format `' @ error/constitute.c/ReadImage/746.
magick: non-conforming drawing primitive definition `text' @ error/draw.c/RenderMVGContent/4548.
`````

**Reply by AW6RO7ZCD** [reply]
if you have installed IM from source [(docs)](https://imagemagick.org/script/install-source.php), try to reinstall it including the following step

`./configure --with-modules
`

also make sure you have installed Ghostscript and Freetype

`sudo apt install ghostscript`

**Comment by snibgo** [comment]
> what is the best way to handle this

For a start, show us the exact command you used. Also show the text output from "magick -version".
